ewms_pilot/pilot.py

```diff
@@ -198,39 +204,91 @@ async def consume_and_reply(
     file_reader: Callable[[Path], Any] = UniversalFileInterface.read,
     #
     debug_dir: Optional[Path] = None,
+    #
+    subproc_timeout: Optional[int] = ENV.EWMS_PILOT_SUBPROC_TIMEOUT,
+    quarantine_time: int = ENV.EWMS_PILOT_QUARANTINE_TIME,
 ) -> None:
     """Communicate with server and outsource processing to subprocesses.
 
     Arguments:
         `timeout_wait_for_first_message`: if None, use 'timeout_incoming'
     """
     LOGGER.info("Making MQClient queue connections...")
-    except_errors = False  # if there's an error, have the cluster try again (probably a system error)
 
     if not queue_incoming or not queue_outgoing:
         raise RuntimeError("Must define an incoming and an outgoing queue")
 
+    try:
+        await _consume_and_reply(
+            cmd,
+            queue_incoming,
+            queue_outgoing,
+            broker_client,
+            broker_address,
+            auth_token,
+            prefetch,
+            timeout_wait_for_first_message,
+            timeout_incoming,
+            timeout_outgoing,
+            fpath_to_subproc,
+            fpath_from_subproc,
+            file_writer,
+            file_reader,
+            debug_dir,
+            subproc_timeout,
+        )
+    except Exception as e:
+        if quarantine_time:
+            LOGGER.error(f"{e} (Quarantining for {quarantine_time} seconds)")
+            await asyncio.sleep(quarantine_time)
+            raise
```

Comment:
I think you want to un-indent this `raise` so it always raises, whether or not it sleeps.